contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,303 @@
+    * @param _bAsset The _bAsset for the liquidation
+    */
+    function deleteLiquidation(address _bAsset)
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+    
+        // Check for unclaimed pTokens and revert
+        // TODO Add mocks to tests for this call
+        //address pToken = IPlatformIntegration(_integration).bAssetToPToken(_bAsset);
+        //uint256 pTokenbalance = IERC20(pToken).balanceOf(address(this));
+        //require(pTokenbalance > 0, "Unclaimed pTokens on this liquidation");
+
+        //Liquidation memory liquidation = liquidations[_bAsset];
+
+        //address sellToken = liquidation.uniswapPath[0];
+        //uint256 sellTokenBalance = IERC20(sellToken).balanceOf(address(this));
+        //require((sellTokenBalance == 0), "Sell tokens on this liquidation");
+
+        delete liquidations[_bAsset];
+        emit LiquidationDeleted(_bAsset);
+    }
+
+    /**
+    * @dev  Update the Uniswap contract address
+    *       Whilst it is unlikely this will be needed it is helpful for testing
+    * @param _uniswapAddress The uniswap address to assign
+    */
+    function updateUniswapAddress(address _uniswapAddress)
+        external
+        onlyGovernor
+    {
+        require(_uniswapAddress != address(0), "_uniswapAddress cannot be zero address");
+        uniswapAddress = _uniswapAddress;
+        emit UniswapUpdated(_uniswapAddress);
+    }
+
+    /**
+    * @dev Triggers a liquidation
+    * @param _bAsset The _bAsset liquidation to be triggered
+    */
+    function triggerLiquidation(address _bAsset)
+        external
+        payable
+    {
+        Liquidation storage liquidation = liquidations[_bAsset];
+        liquidation.lastTriggered = block.timestamp;
+
+        require(block.timestamp > liquidation.lastTriggered.add(1 days),
+                "Trigger liquidation only callable every 24 hours");
+
+        // Token being sold is the first in the Uniswap path
+        address sellToken = liquidation.uniswapPath[0];
+
+        // Token being bought is the last in the Uniswap path
+        address buyToken = liquidation.uniswapPath[liquidation.uniswapPath.length.sub(1)];
+
+        // Get allowance given by the integration contract for the sell token
+        uint256 allowance = IERC20(sellToken).allowance(liquidation.integration, address(this));
+
+        // Transfer tokens to this contract if there is an allowance
+        if (allowance > 0) {
+            IERC20(sellToken).safeTransferFrom(liquidation.integration, address(this), allowance);
+        }
+
+        uint256 balance = IERC20(sellToken).balanceOf(address(this));
+        require((balance > 0), "No sell tokens to liquidate");
+
+        // The amount we want to receive
+        // This computes a randomised amount of the buyToken wanted between 1000 & 4000
+        uint256 randomAmountWanted = uint256(blockhash(block.number-1)).mod(3000).add(1000);
```

Comment:
This will return a number between 1000-4000, which in bAsset units is barely anything.